collections/nemo_asr/nemo_asr/data_layer.py

```diff
@@ -43,31 +55,37 @@ class AudioToTextDataLayer(DataLayerNM):
         int_values (bool): Bool indicating whether the audio file is saved as
             int data or float data.
             Defaults to False.
-        eos_id (str): End of string symbol used for seq2seq models.
+        eos_id (str): Dataset parameter.
+            End of string symbol used for seq2seq models.
             Defaults to None.
-        min_duration (float): All training files which have a duration less
-            than min_duration are dropped. Note: Duration is read from the
-            manifest JSON.
+        min_duration (float): Dataset parameter.
+            All training files which have a duration less than min_duration
+            are dropped. Note: Duration is read from the manifest JSON.
             Defaults to 0.1.
-        max_duration (float): All training files which have a duration more
-            than max_duration are dropped. Note: Duration is read from the
-            manifest JSON.
+        max_duration (float): Dataset parameter.
+            All training files which have a duration more than max_duration
+            are dropped. Note: Duration is read from the manifest JSON.
             Defaults to None.
-        normalize_transcripts (bool): Whether to use automatic text cleaning.
-            It is highly recommended to manually clean text ffor best results.
+        normalize_transcripts (bool): Dataset parameter.
+            Whether to use automatic text cleaning.
+            It is highly recommended to manually clean text for best results.
             Defaults to True.
         trim_silence (bool): Whether to use trim silence from beginning and end
             of audio signal using librosa.effects.trim().
             Defaults to False.
-        load_audio (bool): Controls whether the dataloader loads the audio
-            signal and transcript or just the transcript.
+        load_audio (bool): Dataset parameter.
+            Controls whether the dataloader loads the audio signal and
+            transcript or just the transcript.
             Defaults to True.
         drop_last (bool): See PyTorch DataLoader.
             Defaults to False.
         shuffle (bool): See PyTorch DataLoader.
             Defaults to True.
         num_workers (int): See PyTorch DataLoader.
             Defaults to 0.
+        dataset_type (Dataset, str): The type of dataset used for this
```

Comment:
Isn't AudioDataset necessarily the only dataset that AudioToTextDataLayer currently supports? I'm not sure why we are refactoring to match nemo_nlp since their usecase is different.

Reply:
My reasoning was that if someone has a different dataset format and makes a new dataset for it, this convention makes it easier to extend the loader by just dropping that dataset in.

As far as I could tell, there also seem to be several datasets in nemo_nlp that support a specific dataset, so I figured this was a reasonable small change, but if it just makes things confusing for now I can change it back.

Reply:
Let's pass in a dataset object that the user has to create outside of DataLoader then.
Let's not have DataLoader do it for them.
DataLoader should just check to see if the passed Dataset is a child of a set of acceptable Datasets.

Reply:
Changed this back to a one-to-one correspondence for now as per discussion.